pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotRealtimeTableResource.java

```diff
@@ -104,6 +110,70 @@ public Response pauseConsumption(
     }
   }
 
+  @POST
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/tables/{tableName}/reload")
+  @ApiOperation(value = "Reloads the table across all the servers", notes = "This would reconstruct the table data"
+      + "manager in case of configuration changes. Example usage: trigger after converting the upsert mode"
+      + "from full to partial.")
+  public SuccessResponse reloadTable(
+      @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") @Nullable
+      String tableName,
+      @ApiParam(value = "Whether to force the reload (if table config is not updated, reload will not be done)")
+      @QueryParam("force") boolean force,
+      @Context HttpHeaders httpHeaders, @Context Request request) {
+    String tableNameWithType = TableNameBuilder.REALTIME.tableNameWithType(tableName);
+    validate(tableNameWithType);
+    // checking if another reload job exists
+    // if so, fail the request with the ongoing reload job's id
+    Map<String, Map<String, String>> reloadJobs =
+        _pinotHelixResourceManager.getAllJobsForTable(tableNameWithType,
+            Collections.singleton(ControllerJobType.RELOAD_TABLE));
+    if (reloadJobs.size() > 0) {
+      String jobId = reloadJobs.keySet().iterator().next();
```

Comment:
This doesn't seem correct. We should still all table reload even if the table was reloaded before. We track both the current jobs and the old finished jobs

Reply:
Removed it - thought of avoiding multiple reload jobs at the same time. But looks like it's not possible at the moment because these controller jobs do not have state associated and as you said this gives all the historical + current jobs. So, removed this validation.